src/hotspot/share/utilities/bitMap.cpp

```diff
@@ -706,3 +631,7 @@ void BitMap::print_on(outputStream* st) const {
 }
 
 #endif
+
+template class GrowableBitMap<ArenaBitMap>;
+template class GrowableBitMap<ResourceBitMap>;
+template class GrowableBitMap<CHeapBitMap>;
```

Comment:
Do you really need this explicit template instantiations? I think such code is only required in libraries where you want to be sure to have specific instantiations even though they are not used within the library itself. But here you will get these instantiations anyway whenever `ArenaBitMap`, `ResourceBitMap` or `CHeapBitMap` is used because they are derived from `GrowableBitMap<ArenaBitMap>`, `GrowableBitMap<ResourceBitMap>` and `GrowableBitMap<CHeapBitMap>` respectively.

Reply:
hi, Volker,

Reply:
I'll take my comment back :)

It turns out that we need these explicit instantiations, because implicit instantiations do not instantiate unused members (i.e. "*This applies to the members of the class template: unless the member is used in the program, it is not instantiated, and does not require a definition*", see https://en.cppreference.com/w/cpp/language/class_template#Implicit_instantiation).

E.g.:
```
CHeapBitMap::CHeapBitMap(idx_t size_in_bits, MEMFLAGS flags, bool clear)
  : GrowableBitMap<CHeapBitMap>(), _flags(flags) {
  initialize(size_in_bits, clear);
}
```
implicitely instantiates `GrowableBitMap<CHeapBitMap>` but it will not instantiate `GrowableBitMap<CHeapBitMap>::reallocate()` because it is not used in that compilation unit.